app/services/service_provider_issuer_builder.rb

```diff
@@ -0,0 +1,25 @@
+class ServiceProviderIssuerBuilder
+  ISSUER_TEMPLATE = 'urn:gov:gsa:{protocol}.profiles:sp:sso:{department}:{app}'.freeze
+
+  attr_reader :service_provider
+
+  def initialize(service_provider)
+    @service_provider = service_provider
+  end
+
+  def build_issuer
+    ISSUER_TEMPLATE.gsub('{protocol}', protocol_substring).
```

Comment:
The linter got mad about the `a % b` syntax, but I was able to use `Kernel#format` here.